### ISSUE TITLE
Adding --cluster flag causes traefik to crash

### ISSUE THREAD

**Issue by MadWombat** [containous/traefik#3614]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Added `--cluster` flag to my kubernetes ingress configuration. I tried both with and without `--cluster.node` flag and I tried running locally with `--cluster` on my Mac.

### What did you expect to see?
Traefik starting in cluster mode.

### What did you see instead?
Traefik crashing with a traceback. Tried this on both my kubernetes cluster and on my local OS X system with similar results.

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.6.5-alpine from docker on Kubernetes, 1.6.4 from Homebrew on OS X

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
On kubernetes:
```yml
kind: DaemonSet
apiVersion: extensions/v1beta1
metadata:
  name: traefik-ingress-controller
  namespace: kube-system
  labels:
    app: traefik-ingress-lb
spec:
  minReadySeconds: 5
  updateStrategy:
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: traefik-ingress-lb
        name: traefik-ingress-lb
    spec:
      serviceAccountName: traefik-ingress-controller
      terminationGracePeriodSeconds: 60
      hostNetwork: true
      containers:
      - name: traefik-ingress
        image: traefik:1.6.5-alpine
        ports:
        - name: http
          containerPort: 80
          hostPort: 80
        - name: admin
          containerPort: 8080
        securityContext:
          privileged: true
        args:
        - --api
        - --ping
        - --cluster
        - --consul
        - --consul.endpoint=$(CONSUL_SERVICE_HOST):8500
        - --kubernetes
        - --logLevel=INFO
```

On OS X I just tried to run `traefik --cluster`

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)
kubectl logs
```console
$ kubectl -n kube-system logs traefik-ingress-controller-vx2fm                                                                          
time="2018-07-12T21:08:47Z" level=info msg="Traefik version v1.6.5 built on 2018-07-10_03:54:03PM"                                                                                
time="2018-07-12T21:08:47Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics
/#collected-data\n"                                                                                                                                                               
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                           
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xd9867e]                                                                                                           
                                                                                                                                                                                  
goroutine 1 [running]:                                                                                                                                                            
github.com/containous/traefik/cluster.NewLeadership(0x23c3bc0, 0xc4202e8580, 0xc4202d4900, 0x0)                                                                                   
        /go/src/github.com/containous/traefik/cluster/leadership.go:35 +0x6e                                                                                                      
github.com/containous/traefik/server.NewServer(0xc4200b2ea0, 0x0, 0x100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)                                                                  
        /go/src/github.com/containous/traefik/server/server.go:126 +0x57c                                                                                                         
main.runCmd(0xc4201e4000, 0x0, 0x0)                                                                                                                                               
        /go/src/github.com/containous/traefik/cmd/traefik/traefik.go:180 +0x330                                                                                                   
main.main.func1(0x4808101, 0xc4206149e0)                                                                                                                                          
        /go/src/github.com/containous/traefik/cmd/traefik/traefik.go:54 +0x42                                                                                                     
github.com/containous/traefik/vendor/github.com/containous/staert.(*Staert).Run(0xc42002e920, 0x23a4180, 0xc4206149e0)                                                            
        /go/src/github.com/containous/traefik/vendor/github.com/containous/staert/staert.go:79 +0x2e                                                                              
main.main()                                                                                                                                                                       
        /go/src/github.com/containous/traefik/cmd/traefik/traefik.go:142 +0x1b13                       
```

OS X output
```console
$ traefik --cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x19a362e]

goroutine 1 [running]:
github.com/containous/traefik/cluster.NewLeadership(0x2fd25e0, 0xc4200df600, 0xc420620de0, 0x0)                                                                                   
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/cluster/leadership.go:35 +0x6e                                                               
github.com/containous/traefik/server.NewServer(0xc4204aa350, 0x0, 0x100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)                                                                  
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/server/server.go:125 +0x57c                                                                  
main.runCmd(0xc42020a000, 0x0, 0x0)
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/cmd/traefik/traefik.go:180 +0x330                                                            
main.main.func1(0x2b2a8c0, 0xc420621ea0)
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/cmd/traefik/traefik.go:54 +0x42                                                              
github.com/containous/traefik/vendor/github.com/containous/staert.(*Staert).Run(0xc4206206c0, 0xc42020a000, 0xc420621ea0)                                                         
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/vendor/github.com/containous/staert/staert.go:84 +0x2e                                       
main.main()
        /private/tmp/traefik-20180615-52269-s43ora/src/github.com/containous/traefik/cmd/traefik/traefik.go:142 +0x1b19
```

**Comment by traefiker** [comment]
Closed by #3616.

**Comment by dtomcej** [comment]
@MadWombat Just to give you some clarification on what happened:

The `--cluster` flag was used in the past to ensure that traefik knew it was running in cluster mode. However, over the years, we have changed the code to automatically configure cluster mode if connected to a KV store. Therefore the flag is no longer needed.

The reason that there is no documentation or reference to that flag is that no user configuration is required. Just omit using that flag, and you should be fine.

Thanks!
